index.js

```diff
@@ -52,4 +52,150 @@ const state = {
     }
   ],
   cart: []
-};
\ No newline at end of file
+};
+
+// hint for cart: make a copy of a store item, add a property called
+// quantity
+
+// SELECT EXISTING HTML ELEMENTS
+
+const storeUl = document.querySelector('ul')
+const cartUl = document.querySelector('.cart--item-list-container ul')
+const totalSection = document.querySelector ('.total-section')
+
+//Items in the shop
+function renderStoreItems() {
+ state.items.forEach((item) => {
+  // Create List Item
+  const storeItemli = document.createElement('li')
+  // Store list item in UL
+  storeUl.append(storeItemli)
+  // Create image and save to list item
+  const storeItemImage = document.createElement('img')
+  storeItemImage.setAttribute("src", "assets/icons/" + item.id + ".svg")
+  storeItemli.append(storeItemImage)
+  // Create Button w/ click
+  const addToCartButton = document.createElement('button')
+  addToCartButton.innerText = "Add to cart"
+  storeItemli.append(addToCartButton)
+  addToCartButton.addEventListener('click', function(event){
+    event.preventDefault()
+    addingToCartArray(item)
+  })
+ });
+};
+
+// When button is clicked, pass the item object into this function
+// Is the item already in the array? If yes, increase quantity
+// If not add object into array
+function addingToCartArray(item){
+  const index = state.cart.findIndex((element) => element.id === item.id)
```

Comment:
I should look into this findIndex()... it looks handy for if you don't know exactly which index  the item is at